proc/test_thread_rand.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	unsigned int time;
+	unsigned int id;
+} threads_self_t;
+
+
+typedef struct {
+	unsigned int cthread;
+	unsigned int fin;
+	char stack[10][512];
+	handle_t fin_lock;
+	threads_self_t params[10];
+} threads_common_t;
+
+
+static threads_common_t threads_common = { .cthread = 10, .fin = 0 };
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+
+	mutexLock(threads_common.fin_lock);
+	threads_common.fin++;
+	mutexUnlock(threads_common.fin_lock);
+
+	endthread();
+}
+
+
+TEST_GROUP(threads_rand);
+
+
+TEST_SETUP(threads_rand)
+{
+}
+
+
+TEST_TEAR_DOWN(threads_rand)
+{
+}
+
+
+TEST(threads_rand, test_1)
+{
+	int joined = 0, i;
+
+	srand(7);
+	/* assign threads parameters */
+	for (i = 0; i < threads_common.cthread; i++) {
```

Comment:
could be done in a single `for` loop with starting the thread

Reply:
This loop was divided into two for better clarity, but you are right: one loop is shorter and still clear to read.